src/modules/swap/hooks/useSwapButtonContext.ts

```diff
@@ -64,13 +64,14 @@ export function useSwapButtonContext(input: SwapButtonInput): SwapButtonsContext
   const { isNativeIn, isWrappedOut, wrappedToken } = useDetectNativeToken()
   const isNativeInSwap = isNativeIn && !isWrappedOut
 
-  const nativeInput = trade?.maximumAmountIn(allowedSlippage)
-  const wrapUnwrapAmount = isNativeInSwap ? (nativeInput || parsedAmount)?.wrapped : nativeInput || parsedAmount
+  const wrapUnwrapAmount = isNativeInSwap
+    ? (sellAmountMaxSpend || parsedAmount)?.wrapped
+    : sellAmountMaxSpend || parsedAmount
```

Comment:
I know not part of this change, but can you create a variable for the amount?

```suggestion
  const amount = sellAmountMaxSpend || parsedAmount
  const wrapUnwrapAmount = isNativeInSwap ? (amount)?.wrapped : amount
```

Reply:
Thank you! This looks better